cmp/project.py

```diff
@@ -1014,197 +1037,192 @@ def run_individual(
+        elif dwi_pipeline_config is not None and eeg_pipeline_config is not None and func_pipeline_config is not None:
+            project.dmri_config_file = os.path.abspath(dwi_pipeline_config)
+            project.fmri_config_file = os.path.abspath(func_pipeline_config)
+            project.eeg_config_file = os.path.abspath(eeg_pipeline_config)
+            exit_code = run_dmri_pipeline(project, bids_layout, anat_pipeline)
+            if exit_code == 1:
+                return exit_code
+            exit_code = run_fmri_pipeline(project, bids_layout, anat_pipeline)
+            if exit_code == 1:
+                return exit_code
+            exit_code = run_eeg_pipeline(project, anat_pipeline)
+            if exit_code == 1:
+                return exit_code
+
+    return exit_code
+
+
+def run_anat_pipeline(project, bids_layout, number_of_threads):
+    """Initialize and run anatomical pipeline based on `project.anat_config_file` pipeline configuration, that has to be set a-priori."""
+    anat_pipeline = init_anat_project(project, False)
+    if anat_pipeline is not None:
+        anat_valid_inputs = anat_pipeline.check_input(bids_layout, gui=False)
+
+        print(f"--- Set Freesurfer and ANTs to use {number_of_threads} threads by the means of OpenMP")
+        anat_pipeline.stages["Segmentation"].config.number_of_threads = number_of_threads
+
+        if anat_valid_inputs:
+            print(">> Process anatomical pipeline")
+            anat_pipeline.process()
+            exit_code = 0
+        else:  # pragma: no cover
+            print("ERROR : Invalid inputs for anatomical pipeline")
+            exit_code = 1
+            return exit_code
+
+    anat_valid_outputs, msg = anat_pipeline.check_output()
+    anat_pipeline.fill_stages_outputs()
+
+    return anat_pipeline, exit_code, anat_valid_outputs, msg
+
+
+def run_dmri_pipeline(project, bids_layout, anat_pipeline):
+    """Initialize and run diffusion pipeline based on `project.dmri_config_file` pipeline configuration, that has to be set a-priori."""
+    dmri_valid_inputs, dmri_pipeline = init_dmri_project(
+        project, bids_layout, False
+    )
+    if dmri_pipeline is not None:
+        dmri_pipeline.parcellation_scheme = anat_pipeline.parcellation_scheme
+        dmri_pipeline.atlas_info = anat_pipeline.atlas_info
+        if anat_pipeline.parcellation_scheme == "Custom":
+            dmri_pipeline.custom_atlas_name = anat_pipeline.stages["Parcellation"].config.custom_parcellation.atlas
+            dmri_pipeline.custom_atlas_res = anat_pipeline.stages["Parcellation"].config.custom_parcellation.res
+        if dmri_valid_inputs:
+            dmri_pipeline.process()
+            exit_code = 0
+        else:  # pragma: no cover
+            print("   ... ERROR : Invalid inputs for diffusion pipeline")
+            exit_code = 1
+            return exit_code
+
+        dmri_pipeline.fill_stages_outputs()
+    return exit_code
```

Comment:
`exit_code` may not always be defined.

Reply:
When `if dmri_pipeline is not None:` evaluates to False, there's no exit_code assigned.

Reply:
Addressed by commit https://github.com/connectomicslab/connectomemapper3/pull/149/commits/832eb5bbb29c8b966b8beaf2c2709b71cdc79163